File: project-euler/055/Problem055.hs
revNum :: Integer -> Integer
revNum = read . reverse . show

isPalindromic :: Integer -> Bool
isPalindromic n = n == revNum n

isLychrel :: Integer -> Bool
isLychrel = isL 50
  where isL 0 _ = True
        isL r n = let next = n + revNum n
                      in (not . isPalindromic) next && isL (r - 1) next

main :: IO ()
main = print $ length $ filter isLychrel [0..9999]

File: project-euler/055/problem055.js
function reverse(n){
	return Number(String(n).split('').reverse().join(''));
}

function isPalindromic(n){
	return n == reverse(n);
}

function isLychrel(n, r){
	if(r <= 0){
		return true;
	}
	var next = n + reverse(n);
	return !isPalindromic(next) && isLychrel(next, r - 1);
}

var count = 0;
for(var i = 0; i < 10000; i++){
	if(isLychrel(i, 50)){
		count++;
	}
}
print(count);
